ckanext/datastore/blueprint.py

```diff
@@ -59,6 +61,11 @@ def dump_schema() -> Schema:
 
 
 def dump(resource_id: str):
+    try:
+        resource_id = resource_id_validator(resource_id)  # type: ignore
+    except Invalid:
+        abort(404, _(u'DataStore resource not found'))
```

Comment:
`resource_id_validator` only checks that the input string has valid values and length, not that the resource exists.

If we want to check if a resource has a Datastore table we can do:

```suggestion
    try:
        get_action("datastore_search")({}, {"resource_id": resource_id})
    except ObjectNotFound:
        abort(404, _("DataStore resource not found"))
    
```

Unless there's a better way @wardi ?

Reply:
we could skip the check here and convert NotFound/ValidationErrors from the later datastore_search call to 404's

Reply:
If catching the errors is too late (headers/data have already been sent) then datastore_search here would work but at least add a `"limit": 0` so that it returns more quickly.

Reply:
Did `datastore_search` with limit 0. Also added not supported format response